src/storage/v2/indices/text_index.cpp

```diff
@@ -0,0 +1,470 @@
+          properties[prop_name] = prop_value.ValueInt();
+          break;
+        case PropertyValue::Type::Double:
+          properties[prop_name] = prop_value.ValueDouble();
+          break;
+        case PropertyValue::Type::String:
+          properties[prop_name] = prop_value.ValueString();
+          break;
+        case PropertyValue::Type::Null:
+        case PropertyValue::Type::List:
+        case PropertyValue::Type::Map:
+        case PropertyValue::Type::TemporalData:
+        default:
+          continue;
+      }
+    }
+
+    document["data"] = properties;
+    document["metadata"] = {};
+    document["metadata"]["gid"] = v.gid.AsInt();
+    document["metadata"]["txid"] = -1;
+    document["metadata"]["deleted"] = false;
+    document["metadata"]["is_node"] = true;
+
+    try {
+      mgcxx::text_search::add_document(
+          index_context,
+          mgcxx::text_search::DocumentInput{
+              .data = document.dump(-1, ' ', false, nlohmann::json::error_handler_t::replace)},
+          kDoSkipCommit);
+    } catch (const std::exception &e) {
+      throw query::QueryException(fmt::format("Tantivy error: {}", e.what()));
+    }
+  }
+
+  // As CREATE TEXT INDEX (...) queries don’t accumulate deltas, db_transactional_accessor_->Commit() does not reach
+  // the code area where changes to indices are committed. To get around that without needing to commit text indices
+  // after every such query, we commit here.
+  try {
+    mgcxx::text_search::commit(index_context);
+  } catch (const std::exception &e) {
+    throw query::QueryException(fmt::format("Tantivy error: {}", e.what()));
+  }
+  return true;
+}
+
+bool TextIndex::DropIndex(std::string index_name) {
+  if (!flags::run_time::GetExperimentalTextSearchEnabled()) {
+    throw query::QueryException("To use text indices, enable the text search feature.");
+  }
+
+  try {
+    mgcxx::text_search::drop_index(index_name);
+  } catch (const std::exception &e) {
+    throw query::QueryException(fmt::format("Tantivy error: {}", e.what()));
+  }
+  index_.erase(index_name);
+  std::erase_if(label_to_index_, [index_name](const auto &item) { return item.second == index_name; });
+  return true;
+}
+
+bool TextIndex::IndexExists(std::string index_name) const { return index_.contains(index_name); }
+
+std::vector<Gid> TextIndex::Search(std::string index_name, std::string search_query) {
+  if (!flags::run_time::GetExperimentalTextSearchEnabled()) {
+    throw query::QueryException("To use text indices, enable the text search feature.");
+  }
+
+  if (!index_.contains(index_name)) {
+    throw query::QueryException(fmt::format("Text index \"{}\" doesn’t exist.", index_name));
+  }
+
+  auto input = mgcxx::text_search::SearchInput{.search_query = search_query, .return_fields = {"data", "metadata"}};
+  // Basic check for search fields in the query (Tantivy syntax delimits them with a `:` to the right)
+  if (search_query.find(":") == std::string::npos) {
```

Comment:
can you provide search_fields in constructor

Reply:
That’s not applicable since Tantivy will automatically parse the search fields that are present in the query.